src/Orchard.Cms.Web/Modules/Orchard.Users/Startup.cs

```diff
@@ -28,24 +29,27 @@ public class Startup : StartupBase
         private readonly string _tenantName;
         private readonly string _tenantPrefix;
         private readonly IdentityOptions _options;
+        private readonly IDataProtectionProvider _dataProtectionProvider;
 
-        public Startup(ShellSettings shellSettings, IOptions<IdentityOptions> options)
+        public Startup(ShellSettings shellSettings, IOptions<IdentityOptions> options, IDataProtectionProvider dataProtectionProvider)
         {
             _options = options.Value;
             _tenantName = shellSettings.Name;
             _tenantPrefix = "/" + shellSettings.RequestUrlPrefix;
+            _dataProtectionProvider = dataProtectionProvider.CreateProtector(_tenantName);
         }
 
         public override void Configure(IApplicationBuilder builder, IRouteBuilder routes, IServiceProvider serviceProvider)
         {
             builder.UseIdentity();
-            builder
-                .UseCookieAuthentication(_options.Cookies.ApplicationCookie)
-                .UseCookieAuthentication(_options.Cookies.ExternalCookie)
-                .UseCookieAuthentication(_options.Cookies.TwoFactorRememberMeCookie)
-                .UseCookieAuthentication(_options.Cookies.TwoFactorUserIdCookie)
-                ;
-
+            //We use builder.UseIdentity() instead of commented lines to avoid an unexpected exception. 
```

Comment:
These changes in oauth branch are not needed. Problem that originated this changed was fixed in master adding builder.UseIdentity(), so I will undo this changes on oauth2 branch.